### PR TITLE
Updates to locate parser to include folder contents

### PR DESCRIPTION
**Pull request by sydp**
## One line description of pull request

Updates to the locate parser to include the folder contents from a locate database entry.

## Description:

**Related issue (if applicable):** fixes #<plaso issue number here>

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/main/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [ ] Automated checks (Travis, Codecov, Codefactor )pass
  - [ ] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
  - [ ] Reviewer assigned

### CODE REVIEW

File: tests/parsers/locate.py [hunk 1]
```diff
@@ -34,12 +34,58 @@ def testParseFile(self):
...
         'data_type': 'linux:locate_database:entry',
-        'paths': ['/home/user/temp'],
+        'path': '/home/user/temp',
+        'contents': ['1', '2', '3', 'À'],
         'written_time': '2021-07-09T04:36:19.606373200+00:00'}
 
     event_data = storage_writer.GetAttributeContainerByIndex('event_data', 0)
     self.CheckEventData(event_data, expected_event_values)
 
+    expected_event_values = {
```
Comment (joachimmetz) [inline tests/parsers/locate.py:44]: no need to test all event data objects, just testing individual edge cases should be sufficient

File: plaso/parsers/locate.py [hunk 2]
```diff
@@ -150,7 +152,8 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
         return
 
       event_data = LocateDatabaseEvent()
-      event_data.paths = [directory_header.path]
+      event_data.path = directory_header.path
+      event_data.contents = contents
```
Comment (joachimmetz) [inline plaso/parsers/locate.py:156]: `= contents or None`

File: tests/parsers/locate.py (hunk 1 above)
Comment (sydp) [reply tests/parsers/locate.py:44]: Done

File: plaso/parsers/locate.py (hunk 2 above)
Comment (sydp) [reply plaso/parsers/locate.py]: Done

File: plaso/parsers/locate.py [hunk 3]
```diff
@@ -18,7 +18,8 @@ class LocateDatabaseEvent(events.EventData):
   """Linux locate database (updatedb) event data.
 
   Attributes:
-    paths (list[str]): paths of the locate database (updatedb) entry.
+    contents (list[str]): contents of the locate database (updatedb) entry.
```
Comment (joachimmetz) [inline plaso/parsers/locate.py:21]: Rename contents to entries for consistency with other event data definitions.

File: tests/parsers/locate.py [hunk 4]
```diff
@@ -34,12 +34,31 @@ def testParseFile(self):
...
+        'data_type': 'linux:locate_database:entry',
+        'path': '/home/user/temp/1',
+        'contents': ['1a.txt', '1b.txt'],
+        'written_time': '2021-07-09T04:11:07.438810500+00:00'}
+
+    event_data = storage_writer.GetAttributeContainerByIndex('event_data', 1)
+    self.CheckEventData(event_data, expected_event_values)
+
```
Comment (joachimmetz) [inline tests/parsers/locate.py:52]: Maybe add a comment that you're testing a path without entries

File: tests/parsers/locate.py [hunk 5]
```diff
@@ -34,12 +34,31 @@ def testParseFile(self):
 
     expected_event_values = {
```
Comment (joachimmetz) [inline tests/parsers/locate.py:35]: Maybe add a comment that you're testing a path with entries

**Review by joachimmetz — CHANGES_REQUESTED** [review]
Some small nits, otherwise looks good

**Comment by joachimmetz** [comment]
linter is also warning about:
```
************* Module plaso.parsers.locate
plaso/parsers/locate.py:144:0: C0301: Line too long (81/80) (line-too-long)
ERROR: InvocationError for command /__w/plaso/plaso/.tox/lint/bin/pylint --rcfile=.pylintrc plaso tests tools (exited with code 16)
```

File: tests/parsers/locate.py (hunk 4 above)
Comment (sydp) [reply tests/parsers/locate.py:54]: Done

File: tests/parsers/locate.py (hunk 5 above)
Comment (sydp) [reply tests/parsers/locate.py:36]: Done

File: plaso/parsers/locate.py (hunk 3 above)
Comment (sydp) [reply plaso/parsers/locate.py]: Done

File: data/formatters/linux.yaml [hunk 6]
```diff
@@ -25,8 +25,12 @@ source: 'dpkg log file'
 ---
 type: 'basic'
 data_type: 'linux:locate_database:entry'
-message: 'Folder Paths: {paths}'
-short_message: '{paths}'
+message:
+- 'Folder Path: {path}'
+- 'Contents: {contents}'
```
Comment (joachimmetz) [inline data/formatters/linux.yaml:30]: contents => entries

File: plaso/parsers/locate.py [hunk 7]
```diff
@@ -139,7 +141,8 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
           directory_header.seconds * definitions.NANOSECONDS_PER_SECOND)
 
       try:
-        _, data_size = self._ParseDirectoryEntry(file_object, file_offset)
+        contents, data_size = self._ParseDirectoryEntry(
```
Comment (joachimmetz) [inline plaso/parsers/locate.py:144]: contents => entries

File: tests/parsers/locate.py [hunk 8]
```diff
@@ -32,14 +32,36 @@ def testParseFile(self):
         'recovery_warning')
     self.assertEqual(number_of_warnings, 0)
 
+    # Testing a path containing entries.
     expected_event_values = {
         'data_type': 'linux:locate_database:entry',
-        'paths': ['/home/user/temp'],
+        'path': '/home/user/temp',
```
Comment (joachimmetz) [inline tests/parsers/locate.py:38]: nit: put entries before path

File: data/formatters/linux.yaml (hunk 6 above)
Comment (sydp) [reply data/formatters/linux.yaml]: Done

Comment (sydp) [reply data/formatters/linux.yaml]: Done

Comment (sydp) [reply data/formatters/linux.yaml]: Done

File: plaso/parsers/locate.py (hunk 7 above)
Comment (sydp) [reply plaso/parsers/locate.py]: Done

File: tests/parsers/locate.py (hunk 8 above)
Comment (sydp) [reply tests/parsers/locate.py]: Done

File: plaso/parsers/locate.py [hunk 9]
```diff
@@ -18,7 +18,8 @@ class LocateDatabaseEvent(events.EventData):
   """Linux locate database (updatedb) event data.
 
   Attributes:
-    paths (list[str]): paths of the locate database (updatedb) entry.
+    entries (list[str]): contents of the locate database (updatedb) entry.
+    path: path of the locate database (updatedb) entry.
```
Comment (joachimmetz) [inline plaso/parsers/locate.py:22]: `path:` => `path (str):`

File: data/formatters/linux.yaml [hunk 10]
```diff
@@ -25,8 +25,12 @@ source: 'dpkg log file'
 ---
 type: 'basic'
 data_type: 'linux:locate_database:entry'
-message: 'Folder Paths: {paths}'
-short_message: '{paths}'
+message:
+- 'Folder Path: {path}'
+- 'Entries: {entries}'
+short_message:
+- '{paths}'
```
Comment (joachimmetz) [inline data/formatters/linux.yaml:32]: `paths` => `path`

also keep `paths` for backwards compatibility

File: plaso/parsers/locate.py (hunk 9 above)
Comment (sydp) [reply plaso/parsers/locate.py]: Done

File: data/formatters/linux.yaml (hunk 10 above)
Comment (sydp) [reply data/formatters/linux.yaml]: Done.

Comment (joachimmetz) [reply data/formatters/linux.yaml:35]: I've made some changes

**Review by joachimmetz — APPROVED** [review]
LGTM

**Comment by sydp** [comment]
> LGTM

Thanks for those last couple of updates Joachim.
